gossipd/handshake.c

```diff
@@ -108,20 +83,6 @@ struct act_three {
 /* BOLT #8: The handshake is _exactly_ `66 bytes` */
 #define ACT_THREE_SIZE 66 /* ARM's stupid ABI adds padding. */
 
-static inline void check_act_three(const struct act_three *act3)
-{
-	/* BOLT #8:
-	 *
-	 * `1 byte` for the handshake version, `33 bytes` for the ephemeral
-	 * public key encrypted with the `ChaCha20` stream cipher, `16 bytes`
-	 * for the encrypted public key's tag generated via the `AEAD`
-	 * construction, and `16 bytes` for a final authenticating tag.
-	 */
-	BUILD_ASSERT(sizeof(act3->v) == 1);
-	BUILD_ASSERT(sizeof(act3->ciphertext) == 33 + 16);
-	BUILD_ASSERT(sizeof(act3->tag) == 16);
-}
-
```

Comment:
These functions check that our structures match the requirements; the compilation will fail if this isn't true.

Thus, they don't need to be called, but they *do* need to exist.  Perhaps add a comment instead?

Reply:
@rustyrussell I see. I'll fix that. What would be an appropriate place to call them?

Reply:
I added the calls in #429, seems we had the same idea :-)